### PR TITLE
Support JSR 349 annotations on fields.

### PR DESCRIPTION
**Pull request by azell**
#37 Reasonably clean implementation.

### CODE REVIEW

File: halva-processor/src/main/java/io/soabase/halva/processor/caseclass/Templates.java [hunk 1]
```diff
@@ -638,4 +646,12 @@ private boolean checkParentJsonAnnotations(Element element, FieldSpec.Builder fi
             return false;
         }).count() > 0;
     }
+
+    private boolean checkExternalAnnotation(AnnotationMirror annotation)
```
Comment (Randgalt) [inline halva-processor/src/main/java/io/soabase/halva/processor/caseclass/Templates.java]: What is this method for?

**Comment by azell** [comment]
`checkExternalAnnotation` is meant to prevent attaching duplicate annotations that are already attached to the method, to the field as well.  Internal annotations are considered anything from Jackson or Halva case classes.

For example, if the user had:

```
@JsonProperty("foo_bar")
@Valid
String fooBar();
```

without the check we could have:

```
@JsonProperty("foo_bar")
@Valid
private final String fooBar;

@JsonProperty("foo_bar")
public String fooBar() { return fooBar; }
```

Not sure if it breaks anything, but it seemed a little cleaner.

**Comment by Randgalt** [comment]
checkExternalAnnotation() seems like it can have a cleaner implementation. Maybe a set that captures already created annotations?

**Comment by azell** [comment]
I refactored the validation annotation support to maintain a set of `AnnotationSpec`s that are added to the `fieldBuilder` all at once.

There are also tests to verify that the Jackson annotations are only added once per field definition.  The compiler will complain on duplicates:

```
[ERROR] /private/tmp/halva/halva-test/target/generated-test-sources/test-annotations/io/soabase/halva/caseclass/ParentBeanCase.java:[35,5] com.fasterxml.jackson.annotation.JsonProperty is not a repeatable annotation type
```

**Pull request merged**